src/DIPS.Xamarin.UI/Controls/FloatingActionMenu/FloatingActionMenuBehaviour.cs

```diff
@@ -96,8 +97,23 @@ private static void IsVisiblePropertyChanged(BindableObject bindable, object old
         {
             if (bindable is FloatingActionMenuBehaviour behaviour)
             {
-                behaviour.Children.ForEach(mb => mb.IsVisible = (bool) newvalue);
-                behaviour.Children.ForEach(mb => mb.IsBadgeVisible = (bool)newvalue);
+                behaviour.Children.ForEach(mb => mb.IsVisible = (bool)newvalue);
+
+                if (!(bool)newvalue)
+                {
+                    behaviour.Children.ForEach(mb => mb.IsBadgeVisible = false);
+                }
+                else
+                {
+                    if (behaviour.Children.Any(mb => !string.IsNullOrEmpty(mb.BadgeCount)))
+                    {
+                        behaviour.Children.ForEach(mb => mb.IsBadgeVisible = true);
```

Comment:
Won't the buttons with empty badges also be visible if there is one with a non-empty badgecount?

Reply:
I know this is not the case for us since we only have one badge but this will break if you have more than one badge.

Reply:
Agreed, do it only where the badge is empty.

Reply:
Done